dissect/regf/regf.py

```diff
@@ -478,7 +478,12 @@ def try_decode_sz(data):
 
         # This will return the string utf-16-le decoded up until the first
         # double NULL byte.
-        return data.split(b"\x00\x00")[0].decode("utf-16-le")
+        for idx in range(0, len(data), 2):
+            if data[idx : idx + 2] == b"\x00\x00":
+                break
+        else:
+            idx = len(data)
+        return data[:idx].decode("utf-16-le")
```

Comment:
```suggestion
        idx = -1
        while (idx := data.find(b"\x00\x00", idx + 1)) % 2 == 1:
            if idx == -1:
                idx = len(data)
                break
        return data[:idx].decode("utf-16-le")
```
Would something like this not be a lot faster?

Reply:
I did some tests and it is generally a lot faster except in the worst case scenario (2 zero-bytes starting at odd indexes) where it is ~ 10% slower. But that only starts to become noticeable at larger data sizes (for sizes around 1MB both ways still execute within 0.2 sec).